uri/uri_test.cpp

```diff
@@ -178,5 +182,34 @@ int main() {
         expect_eq(completed, uri::Uri::parse("hax://example.com/test/hello"));
     });
 
+    etest::test("blob URL generation", [] {
+        std::string REGEX_UUID = "[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}";
+
+        uri::Host h = {uri::DNS_DOMAIN, "example.com"};
+        uri::Origin o = {"https", &h, (std::uint16_t)8080, std::nullopt, false};
```

Comment:
Do you really need to cast 8080 to u16 here? If so, use either a static_cast or `std::uint16_t{8080}`.

Reply:
> Do you really need to cast 8080 to u16 here?

IIRC, yes, otherwise msvc complains.